src/pubkey.h

```diff
@@ -271,6 +281,9 @@ class XOnlyPubKey
     /** Construct a Taproot tweaked output point with this point as internal key. */
     std::optional<std::pair<XOnlyPubKey, bool>> CreateTapTweak(const uint256* merkle_root) const;
 
+    /** Tweak an x-only public key by adding the generator multiplied with tweak32 to it. */
+    XOnlyPubKey AddTweak(const unsigned char *tweak32) const;
```

Comment:
Maybe make this a uint256&? Passing a raw pointer without a size (expecting it to be 32 byte) seems a little bug prone.
```suggestion
    XOnlyPubKey AddTweak(const uint256& tweak32) const;
```